tox.ini

```diff
@@ -43,6 +43,8 @@ deps =
     cov: pytest-cov
     jwst: jwst[test] @ git+https://github.com/spacetelescope/jwst.git
     jwst: stpipe @ git+https://github.com/spacetelescope/stpipe.git
+    devdeps: astropy>=0.0.dev0
+    devdeps: numpy>=0.0.dev0
```

Comment:
This will do the same but for `numpy` which I don't think is your intention, this is born out by the fact that `numpy==1.24.3` in these installs. This currently is the latest release of numpy, not a development binary. As @braingram said you should point pip at the anaconda nightly index for this.